test/checkpoint.test.js

```diff
@@ -24,5 +25,51 @@ describe('Checkpoint', function() {
         }
       ], done);
     });
+
+    it('Should be no race condition for current() when calling in parallel', function(done) {
+      async.parallel([
+        function(next) { Checkpoint.current(next); },
+        function(next) { Checkpoint.current(next); }
+      ], function(err, list) {
+        if (err) return done(err);
+        Checkpoint.find(function(err, data) {
+          if (err) return done(err);
+          expect(data).to.have.length(1);
+          done();
+        });
+      });
+    });
+
+    it('Should be no race condition for bumpLastSeq() when calling in parallel', function(done) {
+      async.parallel([
+        function(next) { Checkpoint.bumpLastSeq(next); },
+        function(next) { Checkpoint.bumpLastSeq(next); }
+      ], function(err, list) {
+        if (err) return done(err);
+        Checkpoint.find(function(err, data) {
+          if (err) return done(err);
+          expect(data[0].seq).to.equal(2);
```

Comment:
I am not entirely sure that the current implementation guarantees the value will be exactly `2`, I think it may be possible to arrive at `3`, depending on timing. Perhaps the memory connector guarantees `2`, but I think it won't be the case with external databases.

Could you please add a comment to explain a bit more what we are asserting here?

I would also move the assertion of `data.length` at the top and add a comment too.

We should also check that the values returned by `bumpLastSeq` are all the same (or lower) than the new checkpoint seq.

```js
Checkpoint.find(function(err, data) {
  if (err) return done(err);
  // The invariant "we have at most 1 checkpoint instance" is preserved
  // even when multiple calls are made in parallel
  expect(data).to.have.length(1);

  // There is a race condition here, we could end up with both 2 or 3 as the "seq".
  // The current implementation of the memory connector always yields 2 though.
  expect(data[0].seq).to.equal(2);

  // All checkpoints returned by bumpLastSeq must be 
  //   greater than the old last seq and
  //   lower-or-equal than the new last seq.
  // In this particular case, since the new last seq is always 2, both results
  // should be 2.
  expect(list.map(function(it) { return it.seq; })
    .to.eql([2, 2]);
```